lib/azuremetadata/azuremetadata.py

```diff
@@ -64,10 +74,14 @@ def get_disk_tag(self, device=None):
             print(e, file=sys.stderr)
             return ''
 
+    def show_api_versions(self):
+        data = self._get_api_newest_versions()
```

Comment:
I think the versions `2017-04-02` and `2018-10-01` should be included but I woud like to check if we are
going to treat the new api versions from `_get_api_newest_versions()` as _all_ available API versions or 
not, meaning add those versions as well.

Reply:
Well it appears that treating the error condition as API only returns 3 API versions, in my test these were:

`["2019-11-01","2019-08-15","2019-08-01"]`

Therefore in that test case, `2017-04-02, 2017-08-01, 2017-12-01, 2018-02-01, 2018-04-02, 2018-10-01, 2019-02-01, 2019-03-11, 2019-04-30, 2019-06-01, 2019-06-04` would all have to be hard coded based on [1].

Boy, how hard is it really to understand that when one creates an API that requires a specific version identifier it is also necessary to provide an API that returns those possible version identifiers. It hurts my brain.....

Anyway, I am not a fan of running after Microsoft and more or less constantly updating the list of hard coded metadata API versions. Therefore I'd say we present to the user only what the error returns. We will have some users that complain about this discrepancy but there is nothing we can do other than hard code a list and I am opposed to such an approach.

Meaning we treat whatever the error condition returns as the available APIs, everything else gets ignored. For those users that have the patience or a better memory than I do they can still set the API they want to use even if that particular version does not show up in the list.


[1] https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service

Reply:
Yes, I rather not have it hard coded as well.

Reply:
Done.